underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileSystem.java

```diff
@@ -128,12 +131,35 @@ public void close() throws IOException {
 
   @Override
   public FSDataOutputStream create(String path) throws IOException {
+    return create(path, new UnderFileSystemCreateOptions(mConfiguration));
+  }
+
+  @Override
+  public FSDataOutputStream create(String path, UnderFileSystemCreateOptions options)
+      throws IOException {
     IOException te = null;
     RetryPolicy retryPolicy = new CountingRetry(MAX_TRY);
+    PermissionStatus ps = options.getPermissionStatus();
     while (retryPolicy.attemptRetry()) {
       try {
-        LOG.debug("Creating HDFS file at {}", path);
-        return FileSystem.create(mFileSystem, new Path(path), PERMISSION);
+        LOG.debug("Creating HDFS file at {} with perm {}", path, ps.toString());
+        return FileSystem.create(mFileSystem, new Path(path),
+            new FsPermission(ps.getPermission().toShort()));
+        // TODO(hy): Fix this.
```

Comment:
no need to preserve the commented out code ... instead add a method-level TODO to support block size and replication